src/ui/FormWidgets/TextInput.ts

```diff
@@ -1,14 +1,46 @@
 import { exportGlobally } from '../../GlobalExports';
-import { l } from '../../strings/Strings';
-import { $$ } from '../../utils/Dom';
+import { $$, Dom } from '../../utils/Dom';
 import { KEYBOARD } from '../../utils/KeyboardUtils';
 import { IFormWidget, IFormWidgetSettable } from './FormWidgets';
 
+export interface ITextInputOptions {
+  /**
+   * Uses a placeholder on the input instead of adding a label
+   *
+   * **Default:** `false`
+   */
+  usePlaceholder?: boolean;
+  /**
+   * Set the element class name
+   *
+   * **Default:** `coveo-input`
+   */
+  className?: string;
+  /**
+   * Instead of triggering the `onChange` method by pressing enter or blurring the input,
```

Comment:
Suggestion:

> Whether to trigger the `TextInput`'s `onChange` function on every key press rather than when the `enter` key is pressed or the input is blurred.
>
> **Default:** `false` 